functions/graphql/schema/directives/auth.js

```diff
@@ -19,27 +23,41 @@ module.exports.schema = gql`
 `;
 
 // Adapted from https://www.apollographql.com/docs/graphql-tools/schema-directives.html#Enforcing-access-permissions
+// See also https://www.apollographql.com/docs/graphql-tools/schema-directives.html#Implementing-schema-directives
 class AuthDirective extends SchemaDirectiveVisitor {
+  // Logic to perform when the directive is used on a type
   visitObject(type) {
     this.ensureFieldsWrapped(type);
     type._requiredAuthRoles = this.args.requires;
   }
-  // Visitor methods for nested types like fields and arguments
-  // also receive a details object that provides information about
-  // the parent and grandparent types.
+
+  // Logic to use when the directive is used on a field
+  //  - Visitor methods for nested types like fields and arguments
+  //    also receive a details object that provides information about
+  //    the parent and grandparent types.
   visitFieldDefinition(field, details) {
+    // QUESTION: If one two fields have the directive, will the first one run ensureFIeldsWrapped, mark its object as _authFieldswrapped,
```

Comment:
I don't think that will happen. Instead, if a parent object has a role, and that role is not satisfied, then any of the children properties/objects will not be resolved. That's what that does.

However, if two properties on the same object have different roles, and one is satisfied and the other is not, then the property that is satisfied will go through just fine, while the one that is not will return null